syntax = "proto3";

service TrackerService {
  rpc RegisterPeer (RegisterPeerRequest) returns (RegisterPeerResponse);
  rpc SearchFile (SearchFileRequest) returns (SearchFileResponse);
}

message RegisterPeerRequest {
  string peer_id = 1;
  repeated string files = 2;  // Lista de archivos que el peer tiene
}

message RegisterPeerResponse {
  bool success = 1;
}

message SearchFileRequest {
  string file_name = 1;
}

message SearchFileResponse {
  repeated string peers = 1;  // Lista de peers que tienen el archivo
}
